### DISCUSSION TITLE
Canvas LTI Integration?

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#920] [Q&A]
Curious if anyone has attempted to use this with Canvas and their LTI. I can't find much documentation from Canvas and not sure if there are things I need to get from my campus IT to help facilitate this. 

Are there any difficulties I should be aware of if any of you have done it and gotten it working? 

Got directed here from a defunct gitter page I found. Seemed like this question was asked there at some point but no longer in the historical discussion.

**Comment by anthonygego** [comment]
Hi,

According to the Canvas dev docs, you'll need to define an external app per INGInious task in the Course settings. You may want to follow the 6 steps for LTI1.1 here:  https://github.com/instructure/canvas-lms/blob/master/doc/lti_manual/02_tool_installation.md. You'll probably have to choose *Manual entry* and specify the launch URL (`https://youringinious.instance/lti/<courseid>/<taskid>`), and the couple of LTI key and LTI secret that you specified in your INGInious course.

Then you should be able to see the added app(s) in your Navigation tab. Unfortunately I couldn't deploy a quick instance to check this because their docker compose scripts seem to hang, and there seems to be no public demo.

I hope that can already help. If you manage to configure the task successfully, it would be awesome if you could share the exact procedure.

**Reply by javawolfpack** [reply]
Thanks, and I will update with specifics once I have our INGInious instance setup and can test it. It may be a while as having to wait on campus IT at each step trying to get going. It has taken months to get a managed server that meets the security expectations of campus.
